### PR TITLE
Search by ref

### PR DESCRIPTION
**Pull request by JohnPadgettCivica**
Updates for story 43361.

### CODE REVIEW

File: src/EA.Prsd.Core/Guard.cs [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ﻿namespace EA.Prsd.Core
 {
     using System;
+    using System.Collections.Generic;
+    using System.Linq;
```
Comment (NateHogan) [inline src/EA.Prsd.Core/Guard.cs:5]: Unused Imports

File: src/EA.Weee.DataAccess.Tests.Integration/NotesDataAccessIntegration.cs [hunk 2]
```diff
@@ -5,13 +5,17 @@
     using System.Threading.Tasks;
     using Core.AatfEvidence;
     using Core.Helpers;
```
Comment (NateHogan) [inline src/EA.Weee.DataAccess.Tests.Integration/NotesDataAccessIntegration.cs:7]: Although you didn't add them, these are unused too

File: src/EA.Weee.Requests/AatfEvidence/GetAatfNotesRequest.cs [hunk 3]
```diff
@@ -5,21 +5,26 @@
     using EA.Weee.Core.AatfEvidence;
```
Comment (NateHogan) [inline src/EA.Weee.Requests/AatfEvidence/GetAatfNotesRequest.cs:5]: There is an unused import in the class

File: src/EA.Weee.DataAccess.Tests.Integration/NotesDataAccessIntegration.cs [hunk 4]
```diff
@@ -189,5 +193,99 @@ public async Task GetAllNotes_ShouldNotIncludedSubmitted()
                 notes.Should().NotContain(n => n.Id.Equals(note2.Id));
             }
         }
+
+        [Fact]
+        public async Task GetAllNotes_GivenSearchRefShouldReturnSingleNote()
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.DataAccess.Tests.Integration/NotesDataAccessIntegration.cs:198]: yeah good idea... will add that as a minimum. Will add quite a few more tests once Its available for Sneha.

File: src/EA.Weee.DataAccess/DataAccess/EvidenceNoteFilter.cs [hunk 5]
```diff
@@ -13,5 +14,20 @@ public class EvidenceNoteFilter
...
         public List<NoteStatus> AllowedStatuses { get; set; }
+
+        private string searchRef;
+        public string SearchRef
+        {
+            get
+            {
+                if (!string.IsNullOrWhiteSpace(searchRef))
+                {
+                    return new string(searchRef.Trim().Where(char.IsDigit).ToArray());
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.DataAccess/DataAccess/EvidenceNoteFilter.cs:25]: Yeah this is to cater for them searching for E1 and 1 for example and returning the same record. Dave didn't seem to know how users would use the search so best to cater for both cases. Transfer and evidence notes will always have unique id's so wont hit an issue there.

**Review by NateHogan — APPROVED** [review]

**Review by NateHogan — APPROVED** [review]

**Pull request merged**